src/views/Tree.vue

```diff
@@ -216,14 +218,57 @@ export default {
     }
   },
 
+  setup () {
+    const { optionsArray, getOptions, newOptions } = useViewState()
+    return {
+      optionsArray,
+      getOptions,
+      newOptions,
+      viewStateLocalStore: useViewStateLocalStore()
+    }
+  },
+
   data: () => ({
-    expandAll: null,
-    tasksFilter: {
-      id: null,
-      states: null,
-    },
+    options: {},
   }),
 
```

Comment:
I might be able to move all this into the composable?
https://vuejs.org/guide/reusability/composables